components/automate-cli/pkg/diagnostics/integration/iam_v1_policies.go

```diff
@@ -0,0 +1,146 @@
+package integration
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+
+	"github.com/chef/automate/components/automate-cli/pkg/diagnostics"
+	"github.com/chef/automate/components/automate-cli/pkg/diagnostics/lbrequest"
+)
+
+const createV1PolicyTemplateStr = `
+{
+	"action": "{{ .Action }}",
+	"subjects": ["token:{{ .TokenID }}"],
+	"resource": "{{ .Resource }}"
+}
+`
+
+// PolicyInfo contains info about a created policy
+type PolicyInfo struct {
+	ID string `json:"id"`
+}
+
+type iamV1PoliciesSave struct {
+	TokenID    string `json:"token_id"`
+	TokenValue string `json:"token_value"`
+	PolicyID   string `json:"policy_id"`
+}
+
+// CreateV1PolicyOnToken creates a v1 policy on a given tokenID
+func CreateV1PolicyOnToken(tstCtx diagnostics.TestContext, tokenID string, action string, resource string) (*PolicyInfo, error) {
+	policyInfo := PolicyInfo{}
+	err := MustJSONDecodeSuccess(
+		tstCtx.DoLBRequest(
+			"/api/v0/auth/policies",
+			lbrequest.WithMethod("POST"),
+			lbrequest.WithJSONStringTemplateBody(createV1PolicyTemplateStr, struct {
+				Action   string
+				TokenID  string
+				Resource string
+			}{
+				Action:   action,
+				TokenID:  tokenID,
+				Resource: resource,
+			}),
+		)).WithValue(&policyInfo)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not create policy")
+	}
+
+	return &policyInfo, nil
+}
+
+// DeletePolicy deletes a policy with the given id
+func DeletePolicy(tstCtx diagnostics.TestContext, id string) error {
+	isV2, err := tstCtx.IsIAMV2()
+	if err != nil {
+		return err
+	}
+
+	var reqPath string
+	if isV2 {
+		reqPath = fmt.Sprintf("/apis/iam/v2/policies/%s", id)
+	} else {
+		reqPath = fmt.Sprintf("/api/v0/auth/policies/%s", id)
+	}
+
+	err = MustJSONDecodeSuccess(
+		tstCtx.DoLBRequest(
+			reqPath,
+			lbrequest.WithMethod("DELETE"),
+		)).Error()
+
+	if err != nil {
+		return errors.Wrap(err, "Could not delete policy")
+	}
+	return nil
+}
+
+// CreateIAMV1PoliciesDiagnostic create the diagnostic struct for iam tokens
+// and v1 policies.
+func CreateIAMV1PoliciesDiagnostic() diagnostics.Diagnostic {
```

Comment:
this is basically exactly what the old `auth_policies.go` diagnostic looked like